ui/src/helpers/extract-initials.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (C) 2025 The ORT Server Authors (See <https://github.com/eclipse-apoapsis/ort-server/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+export function extractInitials(fullName: string | undefined): string {
+  // Remove non-alphanumeric characters.
+  const strippedName = (fullName ?? '').replace(/[^a-zA-Z0-9 ]/g, '');
+
+  // Split words, omitting empty strings.
+  const words = strippedName.split(' ').filter(Boolean);
+  if (words.length == 0) return '??';
+
+  const firstName = words.at(0)!;
+  const firstInitial = firstName.at(0)!.toUpperCase();
+
+  if (words.length == 1) return firstInitial;
+
+  const lastName = words.at(-1)!;
+  const lastInitial = lastName.at(0)!.toUpperCase();
```

Comment:
These should check for nullable values and act accordingly instead of using non-null assertion. For index access, `Array.at` seems to be not used as much, so I'd use `array[]` accessor instead as it's much more common, but you can do as you see best.

Reply:
> These should check for nullable values and act accordingly instead of using non-null assertion.

But `lastName` is guaranteed to be non-empty (due to the word splitting) and thus always has a first character. Isn't a non-null assertion the right thing in such cases?

Reply:
> For index access, `Array.at` seems to be not used as much

I was aligning on `at()` as I need to use an index of `-1` in one place, and `[]` does not seem to handle that.

Reply:
> > These should check for nullable values and act accordingly instead of using non-null assertion.
> 
> But `lastName` is guaranteed to be non-empty (due to the word splitting) and thus always has a first character. Isn't a non-null assertion the right thing in such cases?

IMO it's better to tell that to TypeScript with proper non-null handling, as it's easier to see when looking at the code rather than requiring thinking about whether this can be null or not, and if the code above changes later on so that it could be null, non-null assertion still overrides it.

Reply:
I played around with a few syntaxes, but I liked none of them. Unfortunately, something concise like

    const firstName = words[0] && return undefined;

does not work in TypeScript. So I'd actually prefer to keep the non-null assertion in this case.

Reply:
Would this work?

```suggestion
  const firstName = words[0];

  if (!firstName || !firstName[0]) return undefined;

  const firstInitial = firstName[0].toUpperCase();

  const lastName = words[words.length - 1];

  if (!lastName || !lastName[0]) return firstInitial;

  const lastInitial = lastName[0].toUpperCase();
```

Reply:
> Would this work?

Oh it still needs to check for the words.length, or it will give double firstInitial for one word names.

Reply:
Done, please have another look.